modules/autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/jsbridge/response/AutofillResponseWriter.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.jsbridge.response
+
+import com.duckduckgo.autofill.domain.javascript.JavascriptCredentials
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.moshi.Moshi
+import javax.inject.Inject
+
+interface AutofillResponseWriter {
+    fun generateResponseGetAutofillData(credentials: JavascriptCredentials): String
+    fun generateEmptyResponseGetAutofillData(): String
+    fun generateResponseGetAvailableInputTypes(credentialsAvailable: Boolean, emailAvailable: Boolean): String
+    fun generateContentScope(): String
+    fun generateUserUnprotectedDomains(): String
+    fun generateUserPreferences(autofillCredentials: Boolean, showInlineKeyIcon: Boolean = false): String
+}
+
+@ContributesBinding(AppScope::class)
+class AutofillJsonResponseWriter @Inject constructor(val moshi: Moshi) : AutofillResponseWriter {
```

Comment:
Is it worth testing this class?

Reply:
Deferring for now, tracked in https://app.asana.com/0/0/1202592915624853/f